src/index.js

```diff
@@ -62,6 +64,8 @@ export default class ScrollBehavior {
     on(window, 'scroll', this._onWindowScroll);
 
     this._removeTransitionHook = addTransitionHook(() => {
+      this._isTransitioning = true;
```

Comment:
@hedgepigdaniel 

unfortunately, this isn't quite right in all cases.

this `addTransitionHook` API is an old thing meant to allow users to block navigation. transition hooks can return values telling the history API to not allow the navigation (or to rewind the navigation), for things like confirming before leaving unsaved changes on a form

as such, there's no guarantee that transition hook invocation will immediately be followed by a scroll event

what do you think of adding an explicit API to suppress saving scroll position updates? it'd be slightly more work for you, but i think "more explicit" is good.

just add a new method that sets `this._isTransitioning` to `true`. that should make it easier to write a quick test as well

Reply:
Right I see waht you mean (and that explains the `addTransitionHook` API - I always thought that was a bit strange).

Rudy actually has a similar feature, and although I can't think of a good reason to cancel he transition before saving the scroll position, its probably also possible in Rudy to cancel it afterwards.

I'll take your suggestion, that seems reasonable - and also easier to test.

Reply:
The API just matches whatever history had in v1 and v2 when I first wrote this :p

It's useful to block navigation for things like the "dirty form" case. Or really to show a prompt, anyway.

Reply:
in rudy's case, there are multiple middlewares (steps) when changing routes - so I'd expect that such checks for dirty forms etc happen first, and only if the transition is definitely proceeding, then the scroll position is saved (since the checks shouldn't change the page content).

Reply:
This has been implemented